org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/SonarLintImages.java

```diff
@@ -88,22 +88,31 @@ private SonarLintImages() {
   }
 
   /**
-   * Create a composite image with both severity and type
-   * @param type can be null for the header of a groupBy severity
-   *
+   *  Create a composite image with the markers' project connection mode, issue severity and type
+   *  
+   *  @param marker used for finding the project connection mode
+   *  @param severity issue severity
+   *  @param type issue type
+   *  @return composite image if found, null otherwise
    */
   @Nullable
-  public static Image getIssueImage(String severity, @Nullable String type) {
-    var key = severity + "/" + type;
+  public static Image getIssueImage(@Nullable IMarker marker, @Nullable String markerServerKey, String severity, @Nullable String type) {
```

Comment:
Passing the marker here is maybe creating a bit too much coupling. This class is about providing images, and in this case you simply need the "mode"

Reply:
After my previous comment, I think we could rename "mode" to "matchingStatus".